modules/apps/bookmarks/bookmarks-item-selector-web/src/main/java/com/liferay/bookmarks/item/selector/web/internal/BookmarksItemSelectorView.java

```diff
@@ -0,0 +1,145 @@
+ * details.
+ */
+
+package com.liferay.bookmarks.item.selector.web.internal;
+
+import com.liferay.bookmarks.item.selector.criterion.BookmarksItemSelectorCriterion;
+import com.liferay.bookmarks.item.selector.web.internal.constants.BookmarksItemSelectorWebKeys;
+import com.liferay.bookmarks.item.selector.web.internal.display.context.BookmarksItemSelectorViewDisplayContext;
+import com.liferay.bookmarks.service.BookmarksEntryLocalService;
+import com.liferay.item.selector.ItemSelectorReturnType;
+import com.liferay.item.selector.ItemSelectorReturnTypeResolverHandler;
+import com.liferay.item.selector.ItemSelectorView;
+import com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType;
+import com.liferay.item.selector.criteria.URLItemSelectorReturnType;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ListUtil;
+
+import java.io.IOException;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import javax.portlet.PortletURL;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Yang Cao
+ */
+@Component(
+	enabled = false, property = "item.selector.view.order:Integer=100",
+	service = ItemSelectorView.class
+)
+public class BookmarksItemSelectorView
+	implements ItemSelectorView<BookmarksItemSelectorCriterion> {
+
+	@Override
+	public Class<BookmarksItemSelectorCriterion>
+		getItemSelectorCriterionClass() {
+
+		return BookmarksItemSelectorCriterion.class;
+	}
+
+	public ServletContext getServletContext() {
+		return _servletContext;
+	}
+
+	@Override
+	public List<ItemSelectorReturnType> getSupportedItemSelectorReturnTypes() {
+		return _supportedItemSelectorReturnTypes;
+	}
+
+	@Override
+	public String getTitle(Locale locale) {
+		return _language.get(locale, "bookmark-images");
+	}
+
+	@Override
+	public boolean isVisible(ThemeDisplay themeDisplay) {
+		return true;
+	}
+
+	@Override
+	public void renderHTML(
+			ServletRequest servletRequest, ServletResponse servletResponse,
+			BookmarksItemSelectorCriterion bookmarksItemSelectorCriterion,
+			PortletURL portletURL, String itemSelectedEventName, boolean search)
+		throws IOException, ServletException {
+
+		BookmarksItemSelectorViewDisplayContext
+			bookmarksItemSelectorViewDisplayContext =
+				new BookmarksItemSelectorViewDisplayContext(
+					bookmarksItemSelectorCriterion, this,
+					_itemSelectorReturnTypeResolverHandler,
+					itemSelectedEventName, search, portletURL,
+					_bookmarksEntryLocalService);
+
+		servletRequest.setAttribute(
+			BookmarksItemSelectorWebKeys.
+				BOOKMARKS_ITEM_SELECTOR_VIEW_DISPLAY_CONTEXT,
```

Comment:
Instead of adding new constants, use the class name as the request variable name:
```java
		servletRequest.setAttribute(
			BookmarksItemSelectorWebKeys.
				BookmarksItemSelectorViewDisplayContext.class.getName(),
				bookmarksItemSelectorViewDisplayContext);
```

Also, as the variable is used once, you should inline it.

Reply:
This applies to the other display contexts as well.